### PR TITLE
Require site ready

### PR DESCRIPTION
**Pull request by jmendeza**
https://github.com/craftersoftware/craftercms/issues/549
Site lifecycle events
Require Site Ready
InvalidSiteStateException

This PR depends on
https://github.com/craftercms/commons/pull/349

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 1]
```diff
@@ -2109,6 +2109,8 @@ public boolean revertContentItem(@ValidateStringParam String site,
             if (isEmpty(commitId)) {
                 return false;
             }
+            String username = securityService.getCurrentUser();
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java]: Is this related to this issue/PR?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java (hunk 1 above)
Comment (jmendeza) [reply src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java]: @sumerjabri no, it is not. I will exclude from this one

**Review by sumerjabri — APPROVED** [review]
